bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/internal/protocol/CommonCommands.java

```diff
@@ -0,0 +1,460 @@
+        logger.debug("sendBroadcastPing - {} ", macacoToString(macacoFrame));
+        sendBroadcastNow(macacoFrame);
+    }
+
+    /**
+     * Build SUBSCRIPTION Frame
+     */
+    public final void sendSUBSCRIPTIONframe(GatewayConfig gwConfig, int iNodes) {
+        ArrayList<Byte> macacoFrame = new ArrayList<>();
+        macacoFrame.add(SoulissUDPConstants.SOULISS_UDP_FUNCTION_SUBSCRIBE_REQ);
+
+        // PUTIN, STARTOFFEST, NUMBEROF
+        macacoFrame.add((byte) 0x00);// PUTIN
+        macacoFrame.add((byte) 0x00);// PUTIN
+        macacoFrame.add((byte) 0x00);
+
+        macacoFrame.add((byte) iNodes);
+        logger.debug("sendSUBSCRIPTIONframe - {}, IP: {} ", macacoToString(macacoFrame), gwConfig);
+        queueToDispatcher(macacoFrame, gwConfig);
+    }
+
+    /**
+     * Build HEALTHY REQUEST Frame
+     */
+    public final void sendHealthyRequestFrame(GatewayConfig gwConfig, int iNodes) {
+        ArrayList<Byte> macacoFrame = new ArrayList<>();
+        macacoFrame.add(SoulissUDPConstants.SOULISS_UDP_FUNCTION_HEALTHY_REQ);
+
+        // PUTIN, STARTOFFEST, NUMBEROF
+        macacoFrame.add((byte) 0x00);// PUTIN
+        macacoFrame.add((byte) 0x00);// PUTIN
+        macacoFrame.add((byte) 0x00);
+        macacoFrame.add((byte) iNodes);
+        logger.debug("sendHealthyRequestFrame - {}, IP: {} ", macacoToString(macacoFrame), gwConfig);
+        queueToDispatcher(macacoFrame, gwConfig);
+    }
+
+    /**
+     * Build TYPICAL REQUEST Frame
+     */
+    public final void sendTypicalRequestFrame(GatewayConfig gwConfig, int nodes) {
+        ArrayList<Byte> macacoFrame = new ArrayList<>();
+        macacoFrame.add(SoulissUDPConstants.SOULISS_UDP_FUNCTION_TYP_REQ);
+        // PUTIN, STARTOFFEST, NUMBEROF
+        macacoFrame.add((byte) 0x00);// PUTIN
+        macacoFrame.add((byte) 0x00);// PUTIN
+        macacoFrame.add((byte) 0x00); // startOffset
+
+        macacoFrame.add((byte) nodes); // iNodes
+        logger.debug("sendTypicalRequestFrame - {}, IP: {} ", macacoToString(macacoFrame), gwConfig.gatewayLanAddress);
+        queueToDispatcher(macacoFrame, gwConfig);
+    }
+
+    /**
+     * Build TYPICAL REQUEST Frame with start offset
+     */
+    public final void sendTypicalRequestFrame(GatewayConfig gwConfig, int start, int nodes) {
+        ArrayList<Byte> macacoFrame = new ArrayList<>();
+        macacoFrame.add(SoulissUDPConstants.SOULISS_UDP_FUNCTION_TYP_REQ);
+        // PUTIN, STARTOFFEST, NUMBEROF
+        macacoFrame.add((byte) 0x00);// PUTIN
+        macacoFrame.add((byte) 0x00);// PUTIN
+        macacoFrame.add((byte) start); // startOffset
+
+        macacoFrame.add((byte) nodes); // iNodes
+        logger.debug("sendTypicalRequestFrame - {}, IP: {} ", macacoToString(macacoFrame), gwConfig.gatewayLanAddress);
+        queueToDispatcher(macacoFrame, gwConfig);
+    }
+
+    boolean flag = true;
+
+    private final String macacoToString(ArrayList<Byte> mACACOframe) {
```

Comment:
The framework has already a hex converter you could use: `HexUtils`

Reply:
Since macacoFrame is an ArrayList cannot convert with an inline instructions (try to convert arraylist to bytes array)... so i think has not too much sense. What do you think ?

Reply:
OK

Reply:
But, you could use the framework method in `UDPListenDiscoverRunnable`.

Reply:
done